src/Native/dialog.cpp

```diff
@@ -42,4 +55,43 @@ CAMLprim value revery_alert(value vWindow, value vMessage) {
 #endif
   return Val_unit;
 }
+
+CAMLprim value revery_alertOpenFiles(value vStartDirectory, value vFileTypes, value vTitle) {
+  CAMLparam3(vStartDirectory, vFileTypes, vTitle);
+
+  char *startDirectory = NULL;
+
+  char *title = String_val(vTitle);
+
+  if (vStartDirectory != Val_none) {
+    startDirectory = String_val(Some_val(vStartDirectory));
+  }
+  char** fileList;
+#ifdef __APPLE__
+  fileList = revery_open_files_cocoa(startDirectory, NULL, title);
+#endif
+
+  if (fileList) {
+    CAMLlocal2(camlArr, mlData);
+
+    int len = sizeof(fileList) / sizeof(char*);
```

Comment:
```suggestion
    int len = sizeof(fileList) / sizeof(*fileList);
``` less error prone. And change when you change `fileList` type

Reply:
I changed how this worked, so it isn't needed anymore, but thank you!